### ISSUE TITLE
Bit-level parsing seems broken in 6.0.0

### ISSUE THREAD

**Issue by mvforell** [Geal/nom#1231]
## Problem
I'm having problems using the bit-level parses in `nom::bits`. The error message involves `expected an Fn<...> closure, found impl FnMut<...>`. If I recall correctly, one of the changes in 6.0.0 was moving to `FnMut` - maybe you forgot to update something in the `bits` module?

## Environment
- Rust version: 1.47.0
- nom version: 6.0.0

## Test case
```
use nom::{
    bits::{bits, complete::take},
    sequence::tuple,
    IResult
};

fn main() {
    println!("Hello, world!");
}

pub fn example(input: &[u8]) -> IResult<&[u8], (u8, u8)> {
    bits(tuple((
        take(1usize), take(1usize)
    )))(input)
}
```
This gives the following error message:
```
error[E0277]: expected a `std::ops::Fn<((_, usize),)>` closure, found `impl std::ops::FnMut<((_, usize),)>`
  --> src/main.rs:12:10
   |
12 |       bits(tuple((
   |  __________^
13 | |         take(1usize), take(1usize)
14 | |     )))(input)
   | |______^ expected an `Fn<((_, usize),)>` closure, found `impl std::ops::FnMut<((_, usize),)>`
   |
  ::: /home/max/.cargo/registry/src/github.com-1ecc6299db9ec823/nom-6.0.0/src/bits/mod.rs:41:6
   |
41 |     P: Fn((I, usize)) -> IResult<(I, usize), O, E1>,
   |        -------------------------------------------- required by this bound in `nom::bits`
   |
   = help: the trait `std::ops::Fn<((_, usize),)>` is not implemented for `impl std::ops::FnMut<((_, usize),)>`
```

## PS
Also I'm not sure I really understand how the bit-level parsing is supposed to work. I've wrapped my head around the "normal" parsing, but I don't get the return type of `nom::bits::complete::take` and how I'm supposed to use `nom::bits::bits` and `nom::bits::bytes`. More documentation and examples regarding that topic would be really helpful.

**Comment by Leonqn** [comment]
I had the same issue and found a workaround
```
pub fn example(input: &[u8]) -> IResult<&[u8], (u8, u8)> {
    let parser = |i| {
        tuple((
            take::<_, u8, _, nom::error::Error<_>>(1usize),
            take::<_, u8, _, nom::error::Error<_>>(1usize),
        ))(i)
    };
    bits(parser)(input)
}
```
So you just need to create closure for bits parser before passing it to ```bits``` function and some type hints for ```take```.

But still, I also think, that ```bits``` and ```bytes``` combinators should take ```FnMut``` and return ```FnMut```, this will remove this extra closure.

**Comment by mvforell** [comment]
Thank you very much!

I'd also appreciate an appropriate change of `bits` and `bytes`.

**Comment by Geal** [comment]
this should be fixed by #1241, I'll push a release soon
